File: src/main/java/com/viet/news/controller/ChannelController.java
package com.viet.news.controller;

import com.viet.news.entity.ChannelList;
import com.viet.news.repository.ChannelRepository;
import com.viet.news.service.ChannelService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/v1/channel")
public class ChannelController {

    @Autowired
    ChannelService channelService;

    @GetMapping("/allList")
    @ResponseBody
    public List<ChannelList> allList() {
        return channelService.allList();
    }
}

File: src/main/java/com/viet/news/controller/ChannelController.kt
package com.viet.news.controller

import com.viet.news.common.ResponseContent
import com.viet.news.entity.ChannelList
import com.viet.news.repository.ChannelRepository
import com.viet.news.service.ChannelService
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/v1/channel")
class ChannelController : BaseController() {
    @Autowired
    var channelService: ChannelService? = null

    @PostMapping("/list")
    fun list(): ResponseContent<*>? {
        return try {
            ResponseContent.buildSuccess("success", channelService?.list())
        } catch (e: Exception) {
            ResponseContent.buildFail("出错啦\n" + e.message)
        }
    }

    @PostMapping("/allList")
    fun allList(): ResponseContent<*>? {
        return try {
            ResponseContent.buildSuccess("success", channelService?.allList())
        } catch (e: Exception) {
            ResponseContent.buildFail("出错啦\n" + e.message)
        }
    }
}
